scripts/merge_source_cat.py

```diff
@@ -0,0 +1,415 @@
+import os
+import sys
+
+import numpy as np
+import pandas as pd
+
+from lsst.geom import radians
+from lsst.afw.geom.spherePoint import SpherePoint
+
+from lsst.daf.persistence import Butler
+from lsst.daf.persistence.butlerExceptions import NoResults
+
+from astropy.coordinates import SkyCoord, matching
+import astropy.units as u
+
+import GCRCatalogs
+from GCRCatalogs.dc2_source import DC2SourceCatalog
+
+
+class DummyDC2SourceCatalog(GCRCatalogs.BaseGenericCatalog):
+    """
+    A dummy reader class that can be used to generate all native quantities
+    required for the DPDD columns in DC2 Source Catalog
+    """
+    def __init__(self, schema_version=None):
+        self._quantity_modifiers = DC2SourceCatalog._generate_modifiers(dm_schema_version=schema_version)
+
+    @property
+    def required_native_quantities(self):
+        """
+        the set of native quantities that are required by the quantity modifiers
+        """
+        return set(self._translate_quantities(self.list_all_quantities()))
+
+
+def extract_and_save_visit(butler, visit, filename, object_table=None,
+                           dm_schema_version=3,
+                           overwrite=True, verbose=False, **kwargs):
+    """Save catalogs to Parquet from visit-level source catalogs.
+
+    Associates an ObjectID.
+    Iterates through detectors, then saves all to a Parquet file.
+
+    Parameters
+    --
+    butler:  Butler object to use to load
+    visit: int
+        Visit to process
+    filename: str
+        Filename for HDF file.
+    overwrite: bool
+        Overwrite an existing parquet file.
+    """
+    data_refs = butler.subset('src', dataId={'visit': visit})
```

Comment:
Performance wise, is `butler.subset('src', dataId={'visit': visit})` followed by `butler.datasetExists()` on the data refs the same as `butler.queryMetadata(datasetType='src', format=( 'detector', 'filter'), dataId={'visit': visit}`?

Reply:
You still have to check if it exists in either case.  `queryMetadata` is just based on the registry database.  It doesn't check to make sure the product exists.

While `butler.subset` (1 ms) is about twice as slow as `butler.queryMetadata` (0.5 ms), the dominant time usage is the `datasetExists` (250 ms).

Reply:
I see, I wasn't aware of the distinction. Thank you!